source/core/Variants.cpp

```diff
@@ -86,6 +77,21 @@ bool IsTargetTypeCompatibleWithInput(TypeRef inputType, TypeRef targetType, void
     return inputType && (!targetData || inputType->IsA(targetType, true));
 }
 
+void SetVariantToDataTypeError(TypeRef inputType, TypeRef targetType, void* targetData, TypeRef outputType, void* outputData, ErrorCluster* errPtr)
+{
+    if (inputType && inputType->IsCluster() && targetType->IsArray()) {
+        errPtr->SetErrorAndAppendCallChain(true, kUnsupportedOnTarget, "Variant To Data");
+    } else {
+        errPtr->SetErrorAndAppendCallChain(true, kVariantIncompatibleType, "Variant To Data");
+    }
+    if (!targetData
+        && !outputData
+        && IsTargetTypeSameAsOutput(targetType, targetData, outputType, outputData)) {
+        // Emitted via guarantees that  targetData is always the default value of targetType.
+        outputType->CopyData(targetData, outputData);
```

Comment:
Doing InitData here as Craig suggested, will make the DFIR change unnecessary. Please try this suggestion.

Reply:
@spathiwa and @cglzaguilar 

Can I do something like this? If yes, I will likely make this a helper method in TypeAndDataManager.cpp:

`
        TypeRef t = outputType;
        while (t->HasCustomDefault()) {
            t = t->BaseType();
        }
        t->InitData(outputData);
`

Reply:
Went back to using `initData`. We should be protected in this case, because the output terminal should always be a local with the default-default value in via. That said, I am filing an issue to come up with a `InitDefaultData` that does what I described above. The reason I am not doing it now is that I would like this API to have dedicated tests, and I don't have the time for it at the moment.